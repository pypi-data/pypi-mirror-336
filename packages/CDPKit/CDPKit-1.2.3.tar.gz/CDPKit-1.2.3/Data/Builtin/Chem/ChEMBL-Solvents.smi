WATER	[OH2]
DICHLOROMETHANE	ClCCl
TRICHLOROMETHANE	ClC(Cl)Cl
ETHYL ACETATE	CCOC(=O)C
METHANOL	CO
PROPAN-2-OL	CC(C)O
ACETONE	CC(=O)C
DMSO	CS(=O)C
ETHANOL	CCO
