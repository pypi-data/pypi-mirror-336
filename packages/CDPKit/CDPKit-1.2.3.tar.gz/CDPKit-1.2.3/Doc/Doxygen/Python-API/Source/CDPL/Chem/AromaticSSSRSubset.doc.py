#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) Thomas Seidel <thomas.seidel@univie.ac.at>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2 of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this program; see the file COPYING. If not, write to
# the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
# Boston, MA 02111-1307, USA.
#

##
# \brief Implements the extraction of the aromatic rings in the SSSR of a molecular graph.
# 
class AromaticSSSRSubset(FragmentList):

    ##
    # \brief Constructs an empty <tt>AromaticSSSRSubset</tt> instance.
    # 
    def __init__() -> None: pass

    ##
    # \brief Initializes a copy of the \e %AromaticSSSRSubset instance \a arom_sssr.
    # \param arom_sssr The \e %AromaticSSSRSubset instance to copy.
    # 
    def __init__(arom_sssr: AromaticSSSRSubset) -> None: pass

    ##
    # \brief Construct a <tt>AromaticSSSRSubset</tt> instance that contains all aromatic rings of the SSSR of the molecular graph <em>molgraph</em>.
    # 
    # \param molgraph The molecular graph for which to extract the aromatic rings.
    # 
    def __init__(molgraph: MolecularGraph) -> None: pass

    ##
    # \brief Replaces the current state of \a self with a copy of the state of the \c %AromaticSSSRSubset instance \a arom_sssr.
    # \param arom_sssr The \c %AromaticSSSRSubset instance to copy.
    # \return \a self
    # 
    def assign(arom_sssr: AromaticSSSRSubset) -> AromaticSSSRSubset: pass

    ##
    # \brief Replaces the current set of rings by the aromatic rings in the SSSR of the molecular graph <em>molgraph</em>.
    # 
    # \param molgraph The molecular graph for which to extract the aromatic rings.
    # 
    def extract(molgraph: MolecularGraph) -> None: pass
