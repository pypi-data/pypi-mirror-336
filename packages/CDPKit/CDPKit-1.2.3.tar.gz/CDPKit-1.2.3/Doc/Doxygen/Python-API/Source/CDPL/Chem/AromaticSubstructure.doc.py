#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) Thomas Seidel <thomas.seidel@univie.ac.at>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2 of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this program; see the file COPYING. If not, write to
# the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
# Boston, MA 02111-1307, USA.
#

##
# \brief Implements the perception of aromatic atoms and bonds in a molecular graph.
# 
class AromaticSubstructure(Fragment):

    ##
    # \brief Constructs an empty <tt>AromaticSubstructure</tt> instance.
    # 
    def __init__() -> None: pass

    ##
    # \brief Construct a <tt>AromaticSubstructure</tt> instance that consists of the aromatic atoms and bonds of the molecular graph <em>molgraph</em>.
    # 
    # \param molgraph The molecular graph for which to perceive aromatic atoms and bonds.
    # 
    def __init__(molgraph: MolecularGraph) -> None: pass

    ##
    # \brief Replaces the currently stored atoms and bonds by the set of aromatic atoms and bonds of the molecular graph <em>molgraph</em>.
    # 
    # \param molgraph The molecular graph for which to perceive aromatic atoms and bonds.
    # 
    def perceive(molgraph: MolecularGraph) -> None: pass
