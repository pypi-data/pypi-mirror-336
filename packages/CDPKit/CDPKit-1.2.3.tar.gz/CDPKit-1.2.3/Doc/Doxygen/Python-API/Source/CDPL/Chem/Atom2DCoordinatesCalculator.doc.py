#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) Thomas Seidel <thomas.seidel@univie.ac.at>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2 of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this program; see the file COPYING. If not, write to
# the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
# Boston, MA 02111-1307, USA.
#

##
# \brief Atom2DCoordinatesCalculator.
# 
class Atom2DCoordinatesCalculator(Boost.Python.instance):

    ##
    # \brief Constructs the <tt>Atom2DCoordinatesCalculator</tt> instance.
    # 
    def __init__() -> None: pass

    ##
    # \brief Constructs the <tt>Atom2DCoordinatesCalculator</tt> instance and calculates 2D-coordinates for the atoms of the molecular graph <em>molgraph</em>.
    # 
    # \param molgraph The molecular graph for which to calculate 2D-coordinates.
    # \param coords An array containing the calculated 2D-coordinates. The coordinates are stored in the same order as the atoms appear in the atom list of the molecular graph (i.e. the coordinates of an atom are accessible via its index).
    # 
    def __init__(molgraph: MolecularGraph, coords: Math.Vector2DArray) -> None: pass

    ##
    # \brief Returns the numeric identifier (ID) of the wrapped C++ class instance.
    # 
    # Different Python \c %Atom2DCoordinatesCalculator instances may reference the same underlying C++ class instance. The commonly used Python expression
    # <tt>a is not b</tt> thus cannot tell reliably whether the two \c %Atom2DCoordinatesCalculator instances \e a and \e b reference different C++ objects. 
    # The numeric identifier returned by this method allows to correctly implement such an identity test via the simple expression
    # <tt>a.getObjectID() != b.getObjectID()</tt>.
    # 
    # \return The numeric ID of the internally referenced C++ class instance.
    # 
    def getObjectID() -> int: pass

    ##
    # \brief Calculates 2D-coordinates for the atoms of the molecular graph <em>molgraph</em>.
    # 
    # \param molgraph The molecular graph for which to calculate 2D-coordinates.
    # \param coords An array containing the calculated 2D-coordinates. The coordinates are stored in the same order as the atoms appear in the atom list of the molecular graph (i.e. the coordinates of an atom are accessible via its index).
    # 
    def calculate(molgraph: MolecularGraph, coords: Math.Vector2DArray) -> None: pass

    objectID = property(getObjectID)
