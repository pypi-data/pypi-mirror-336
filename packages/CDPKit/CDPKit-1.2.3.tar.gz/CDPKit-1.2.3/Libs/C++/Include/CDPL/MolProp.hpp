/* 
 * MolProp.hpp 
 *
 * This file is part of the Chemical Data Processing Toolkit
 *
 * Copyright (C) 2003 Thomas Seidel <thomas.seidel@univie.ac.at>
 *
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 2 of the License, or (at your option) any later version.
 *
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this library; see the file COPYING. If not, write to
 * the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
 * Boston, MA 02111-1307, USA.
 */

/**
 * \file
 * \brief A convenience header including everything that is defined in namespace CDPL::MolProp.
 */

#ifndef CDPL_MOLPROP_HPP
#define CDPL_MOLPROP_HPP

#include "CDPL/MolProp/AtomProperty.hpp"
#include "CDPL/MolProp/BondProperty.hpp"
#include "CDPL/MolProp/AtomPropertyDefault.hpp"
#include "CDPL/MolProp/MassComposition.hpp"
#include "CDPL/MolProp/ElementHistogram.hpp"
#include "CDPL/MolProp/CoordinationGeometry.hpp"
#include "CDPL/MolProp/HBondDonorAtomType.hpp"
#include "CDPL/MolProp/HBondAcceptorAtomType.hpp"
#include "CDPL/MolProp/AtomFunctions.hpp"
#include "CDPL/MolProp/BondFunctions.hpp"
#include "CDPL/MolProp/AtomContainerFunctions.hpp"
#include "CDPL/MolProp/BondContainerFunctions.hpp"
#include "CDPL/MolProp/MolecularGraphFunctions.hpp"
#include "CDPL/MolProp/TPSACalculator.hpp"
#include "CDPL/MolProp/AtomHydrophobicityCalculator.hpp"
#include "CDPL/MolProp/LogSCalculator.hpp"
#include "CDPL/MolProp/XLogPCalculator.hpp"
#include "CDPL/MolProp/PEOESigmaChargeCalculator.hpp"
#include "CDPL/MolProp/MHMOPiChargeCalculator.hpp"
#include "CDPL/MolProp/HBondDonorAtomTyper.hpp"
#include "CDPL/MolProp/HBondAcceptorAtomTyper.hpp"

#endif // CDPL_MOLPROP_HPP
