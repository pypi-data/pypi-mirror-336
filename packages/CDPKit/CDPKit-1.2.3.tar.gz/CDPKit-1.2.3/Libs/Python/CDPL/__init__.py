##
# __init__.py  
#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) 2003 Thomas Seidel <thomas.seidel@univie.ac.at>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2 of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this program; see the file COPYING. If not, write to
# the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
# Boston, MA 02111-1307, USA.
##


from __future__ import absolute_import

import os
import sys
import platform

if sys.version_info[1] >= 8 and platform.system() == 'Windows':
    if 'site-packages' in __file__:
        os.add_dll_directory(os.path.dirname(__file__))
    else:
        os.add_dll_directory(os.path.join(os.path.dirname(os.path.dirname(os.path.dirname(__file__))), '@CDPKIT_EXECUTABLE_INSTALL_DIR@'))

from ._cdpl import *

__version__ = VERSION_STRING
