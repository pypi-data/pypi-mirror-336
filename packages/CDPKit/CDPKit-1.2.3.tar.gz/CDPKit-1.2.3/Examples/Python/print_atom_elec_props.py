#!/bin/env python

##
# print_atom_elec_props.py 
#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) 2003 Thomas Seidel <thomas.seidel@univie.ac.at>
#
# Permission to use, copy, modify, and/or distribute this software for any purpose with
# or without fee is hereby granted.
# THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH REGARD
# TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS. IN
# NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL
# DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER
# IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
# CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
##


import sys
import os

import CDPL.Chem as Chem
import CDPL.MolProp as MolProp


# outputs the corresponding properties of each atom of the provided molecule
def outputProperties(mol: Chem.Molecule) -> None:
    Chem.calcBasicProperties(mol, False)       # calculate basic molecular properties (if not yet done)
    Chem.perceivePiElectronSystems(mol, False) # perceive pi electron systems and store info as Chem.MolecularGraph property
                                               # (required for MHMO calculations)

    # calculate sigma charges and electronegativities using the PEOE method and store values as atom properties
    # (prerequisite for MHMO calculations)
    MolProp.calcPEOEProperties(mol, False)  

    # calculate pi charges, electronegativities and other properties by a modified Hueckel MO method and store values as properties
    MolProp.calcMHMOProperties(mol, False)
                
    for atom in mol.atoms:
        print('- Atom #%s' % str(atom.getIndex()))
        print('\tSigma charge: %s' % str(MolProp.getPEOESigmaCharge(atom)))
        print('\tPi charge: %s' % str(MolProp.getMHMOPiCharge(atom)))
        print('\tTotal partial charge: %s' % str(MolProp.calcTotalPartialCharge(atom)))
        print('\tLone-pair electronegativity: %s' % str(MolProp.calcLonePairElectronegativity(atom, mol)))
        print('\tPi electronegativity: %s' % str(MolProp.calcPiElectronegativity(atom, mol)))
        print('\tSigma electronegativity: %s' % str(MolProp.getPEOESigmaElectronegativity(atom)))
        print('\tExerted inductive effect: %s' % str(MolProp.calcInductiveEffect(atom, mol)))
        print('\tFree valence electron count: %s' % str(MolProp.calcFreeValenceElectronCount(atom, mol)))
        print('\tValence electron count: %s' % str(MolProp.calcValenceElectronCount(atom)))

def main() -> None:
    if len(sys.argv) < 2:
        sys.exit('Usage: %s <input mol. file>' % sys.argv[0])

    # create reader for input molecules (format specified by file extension)
    reader = Chem.MoleculeReader(sys.argv[1]) 
   
    # create an instance of the default implementation of the Chem.Molecule interface
    mol = Chem.BasicMolecule()
    
    # read and process molecules one after the other until the end of input has been reached
    try:
        while reader.read(mol):
            try:
                outputProperties(mol)
            except Exception as e:
                sys.exit('Error: processing of molecule failed: ' + str(e))
                
    except Exception as e: # handle exception raised in case of severe read errors
        sys.exit('Error: reading molecule failed: ' + str(e))

    sys.exit(0)
        
if __name__ == '__main__':
    main()
