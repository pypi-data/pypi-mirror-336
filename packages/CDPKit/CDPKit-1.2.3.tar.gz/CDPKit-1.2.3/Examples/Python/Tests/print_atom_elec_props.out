- Atom #0
	Sigma charge: -0.19743878532183057
	Pi charge: 0.033927880351130923
	Total partial charge: -0.16351090497069964
	Lone-pair electronegativity: 5.036054989879319
	Pi electronegativity: 6.375440171650052
	Sigma electronegativity: 12.01552141411567
	Exerted inductive effect: 25.034199940915727
	Free valence electron count: 6
	Valence electron count: 7
- Atom #1
	Sigma charge: 0.12413716570239582
	Pi charge: 0.016671669815045975
	Total partial charge: 0.14080883551744178
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 5.743026500571142
	Sigma electronegativity: 9.97022753856835
	Exerted inductive effect: 51.61760965119876
	Free valence electron count: 0
	Valence electron count: 4
- Atom #2
	Sigma charge: -0.02674803476341382
	Pi charge: -0.021837739725420136
	Total partial charge: -0.04858577448883396
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 5.4151261275585325
	Sigma electronegativity: 8.541788656624178
	Exerted inductive effect: 51.69759578851705
	Free valence electron count: 0
	Valence electron count: 4
- Atom #3
	Sigma charge: -0.05136368283743562
	Pi charge: 0.005103678338195028
	Total partial charge: -0.046260004499240594
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 5.643610955970588
	Sigma electronegativity: 8.315274200106183
	Exerted inductive effect: 49.71438534287647
	Free valence electron count: 0
	Valence electron count: 4
- Atom #4
	Sigma charge: -0.011154722829764789
	Pi charge: -0.017131427391726683
	Total partial charge: -0.028286150221491474
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 5.454731772613736
	Sigma electronegativity: 8.686225869267117
	Exerted inductive effect: 43.02259104841501
	Free valence electron count: 0
	Valence electron count: 4
- Atom #5
	Sigma charge: 0.124463030542395
	Pi charge: 0.0
	Total partial charge: 0.124463030542395
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 9.151693786806165
	Exerted inductive effect: 48.65454454958518
	Free valence electron count: 0
	Valence electron count: 4
- Atom #6
	Sigma charge: -0.350583435249159
	Pi charge: 0.0
	Total partial charge: -0.350583435249159
	Lone-pair electronegativity: 3.577313399459457
	Pi electronegativity: 3.577313399459457
	Sigma electronegativity: 9.821305172229694
	Exerted inductive effect: 42.30417192780788
	Free valence electron count: 4
	Valence electron count: 6
- Atom #7
	Sigma charge: 0.07851861227006815
	Pi charge: 0.0
	Total partial charge: 0.07851861227006815
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 8.712391384888123
	Exerted inductive effect: 46.67178322584794
	Free valence electron count: 0
	Valence electron count: 4
- Atom #8
	Sigma charge: -0.014431328341953275
	Pi charge: -0.0017119886912542182
	Total partial charge: -0.016143317033207493
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 5.585405353325122
	Sigma electronegativity: 8.655814497341941
	Exerted inductive effect: 60.71659855072372
	Free valence electron count: 0
	Valence electron count: 4
- Atom #9
	Sigma charge: -0.006535262406970195
	Pi charge: 0.016512635272432075
	Total partial charge: 0.00997737286546188
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 5.741654420217348
	Sigma electronegativity: 8.729155845945677
	Exerted inductive effect: 56.76719842027073
	Free valence electron count: 0
	Valence electron count: 4
- Atom #10
	Sigma charge: -0.05232306087259815
	Pi charge: 0.002605376914421864
	Total partial charge: -0.04971768395817629
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 5.6222438217672845
	Sigma electronegativity: 8.306483003742992
	Exerted inductive effect: 63.91579227723197
	Free valence electron count: 0
	Valence electron count: 4
- Atom #11
	Sigma charge: -0.043167103902568935
	Pi charge: 0.02428141485036539
	Total partial charge: -0.018885689052203544
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 5.808853854768595
	Sigma electronegativity: 8.390496323905653
	Exerted inductive effect: 66.45517671190235
	Free valence electron count: 0
	Valence electron count: 4
- Atom #12
	Sigma charge: 0.05043634546594092
	Pi charge: -0.017815844724915486
	Total partial charge: 0.03262050074102543
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 5.4489640132068615
	Sigma electronegativity: 9.263907915407946
	Exerted inductive effect: 52.41594424609129
	Free valence electron count: 0
	Valence electron count: 4
- Atom #13
	Sigma charge: -0.03846905917894388
	Pi charge: 0.023636323829940742
	Total partial charge: -0.014832735349003139
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 5.8032603491337245
	Sigma electronegativity: 8.433702969908552
	Exerted inductive effect: 66.80765696672404
	Free valence electron count: 0
	Valence electron count: 4
- Atom #14
	Sigma charge: 0.06033576812192752
	Pi charge: 0.0
	Total partial charge: 0.06033576812192752
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.544456566328504
	Exerted inductive effect: 37.889269434966394
	Free valence electron count: 0
	Valence electron count: 1
- Atom #15
	Sigma charge: 0.09723071103211513
	Pi charge: 0.25364728489338406
	Total partial charge: 0.3508779959254992
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 7.952761958833895
	Sigma electronegativity: 11.31573150140599
	Exerted inductive effect: 43.35002484102831
	Free valence electron count: 0
	Valence electron count: 4
- Atom #16
	Sigma charge: -0.18263839225070008
	Pi charge: -0.30115520234437576
	Total partial charge: -0.48379359459507587
	Lone-pair electronegativity: 2.235588683595788
	Pi electronegativity: 6.261914078439771
	Sigma electronegativity: 13.53412447943935
	Exerted inductive effect: 53.09252802302599
	Free valence electron count: 2
	Valence electron count: 5
- Atom #17
	Sigma charge: 0.06005769882873894
	Pi charge: 0.0
	Total partial charge: 0.06005769882873894
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.542740161465713
	Exerted inductive effect: 37.088953561726065
	Free valence electron count: 0
	Valence electron count: 1
- Atom #18
	Sigma charge: 0.059347266257900426
	Pi charge: 0.0
	Total partial charge: 0.059347266257900426
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.538354566562418
	Exerted inductive effect: 36.911626490072145
	Free valence electron count: 0
	Valence electron count: 1
- Atom #19
	Sigma charge: 0.056757588314477504
	Pi charge: 0.0
	Total partial charge: 0.056757588314477504
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.522363353736825
	Exerted inductive effect: 28.50769186968725
	Free valence electron count: 0
	Valence electron count: 1
- Atom #20
	Sigma charge: 0.056757588314477504
	Pi charge: 0.0
	Total partial charge: 0.056757588314477504
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.522363353736825
	Exerted inductive effect: 28.50769186968725
	Free valence electron count: 0
	Valence electron count: 1
- Atom #21
	Sigma charge: -0.015391279464276772
	Pi charge: 0.0
	Total partial charge: -0.015391279464276772
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.839153410507009
	Exerted inductive effect: 30.568807115041455
	Free valence electron count: 0
	Valence electron count: 4
- Atom #22
	Sigma charge: -0.036608456572135926
	Pi charge: 0.0
	Total partial charge: -0.036608456572135926
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.646453905361869
	Exerted inductive effect: 18.42149549202137
	Free valence electron count: 0
	Valence electron count: 4
- Atom #23
	Sigma charge: 0.0023920958722880533
	Pi charge: 0.0
	Total partial charge: 0.0023920958722880533
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 8.00197019769821
	Exerted inductive effect: 11.244714671478935
	Free valence electron count: 0
	Valence electron count: 4
- Atom #24
	Sigma charge: -0.3007851968667249
	Pi charge: 0.0
	Total partial charge: -0.3007851968667249
	Lone-pair electronegativity: 1.6349406628290533
	Pi electronegativity: 1.6349406628290533
	Sigma electronegativity: 8.408545729031685
	Exerted inductive effect: 6.711675472161389
	Free valence electron count: 2
	Valence electron count: 5
- Atom #25
	Sigma charge: -0.009572515364496574
	Pi charge: 0.0
	Total partial charge: -0.009572515364496574
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.892296579088681
	Exerted inductive effect: 3.327917309106125
	Free valence electron count: 0
	Valence electron count: 4
- Atom #26
	Sigma charge: 0.037009538544604405
	Pi charge: 0.0
	Total partial charge: 0.037009538544604405
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.400172485190092
	Exerted inductive effect: 1.7744874824867545
	Free valence electron count: 0
	Valence electron count: 1
- Atom #27
	Sigma charge: 0.037009538544604405
	Pi charge: 0.0
	Total partial charge: 0.037009538544604405
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.400172485190092
	Exerted inductive effect: 1.7744874824867545
	Free valence electron count: 0
	Valence electron count: 1
- Atom #28
	Sigma charge: 0.037009538544604405
	Pi charge: 0.0
	Total partial charge: 0.037009538544604405
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.400172485190092
	Exerted inductive effect: 1.7744874824867545
	Free valence electron count: 0
	Valence electron count: 1
- Atom #29
	Sigma charge: -0.009572515364496574
	Pi charge: 0.0
	Total partial charge: -0.009572515364496574
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.892296579088681
	Exerted inductive effect: 3.327917309106125
	Free valence electron count: 0
	Valence electron count: 4
- Atom #30
	Sigma charge: 0.037009538544604405
	Pi charge: 0.0
	Total partial charge: 0.037009538544604405
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.400172485190092
	Exerted inductive effect: 1.7744874824867545
	Free valence electron count: 0
	Valence electron count: 1
- Atom #31
	Sigma charge: 0.037009538544604405
	Pi charge: 0.0
	Total partial charge: 0.037009538544604405
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.400172485190092
	Exerted inductive effect: 1.7744874824867545
	Free valence electron count: 0
	Valence electron count: 1
- Atom #32
	Sigma charge: 0.037009538544604405
	Pi charge: 0.0
	Total partial charge: 0.037009538544604405
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.400172485190092
	Exerted inductive effect: 1.7744874824867545
	Free valence electron count: 0
	Valence electron count: 1
- Atom #33
	Sigma charge: 0.04025372793740587
	Pi charge: 0.0
	Total partial charge: 0.04025372793740587
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.420275859266211
	Exerted inductive effect: 6.713741176906788
	Free valence electron count: 0
	Valence electron count: 1
- Atom #34
	Sigma charge: 0.04025372793740587
	Pi charge: 0.0
	Total partial charge: 0.04025372793740587
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.420275859266211
	Exerted inductive effect: 6.713741176906788
	Free valence electron count: 0
	Valence electron count: 1
- Atom #35
	Sigma charge: 0.026733226209445057
	Pi charge: 0.0
	Total partial charge: 0.026733226209445057
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.33641511893214
	Exerted inductive effect: 10.880312441723596
	Free valence electron count: 0
	Valence electron count: 1
- Atom #36
	Sigma charge: 0.026733226209445057
	Pi charge: 0.0
	Total partial charge: 0.026733226209445057
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.33641511893214
	Exerted inductive effect: 10.880312441723596
	Free valence electron count: 0
	Valence electron count: 1
- Atom #37
	Sigma charge: 0.028489268827862953
	Pi charge: 0.0
	Total partial charge: 0.028489268827862953
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.347318519960392
	Exerted inductive effect: 17.994319645074224
	Free valence electron count: 0
	Valence electron count: 1
- Atom #38
	Sigma charge: 0.028489268827862953
	Pi charge: 0.0
	Total partial charge: 0.028489268827862953
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.347318519960392
	Exerted inductive effect: 17.994319645074224
	Free valence electron count: 0
	Valence electron count: 1
- Atom #39
	Sigma charge: -0.05136368283743562
	Pi charge: 0.005103678338194917
	Total partial charge: -0.046260004499240705
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 5.643610955970587
	Sigma electronegativity: 8.315274200106183
	Exerted inductive effect: 49.71438534287647
	Free valence electron count: 0
	Valence electron count: 4
- Atom #40
	Sigma charge: -0.02674803476341382
	Pi charge: -0.021837739725420136
	Total partial charge: -0.04858577448883396
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 5.4151261275585325
	Sigma electronegativity: 8.541788656624178
	Exerted inductive effect: 51.697595788517056
	Free valence electron count: 0
	Valence electron count: 4
- Atom #41
	Sigma charge: 0.0613478704318821
	Pi charge: 0.0
	Total partial charge: 0.0613478704318821
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.550703117219289
	Exerted inductive effect: 28.586137318891808
	Free valence electron count: 0
	Valence electron count: 1
- Atom #42
	Sigma charge: 0.05937763019238762
	Pi charge: 0.0
	Total partial charge: 0.05937763019238762
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.5385420187388315
	Exerted inductive effect: 28.516928655507627
	Free valence electron count: 0
	Valence electron count: 1
- Atom #43
	Sigma charge: 0.05937763019238762
	Pi charge: 0.0
	Total partial charge: 0.05937763019238762
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.5385420187388315
	Exerted inductive effect: 28.516928655507623
	Free valence electron count: 0
	Valence electron count: 1
- Atom #44
	Sigma charge: 0.0613478704318821
	Pi charge: 0.0
	Total partial charge: 0.0613478704318821
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.550703117219289
	Exerted inductive effect: 28.586137318891804
	Free valence electron count: 0
	Valence electron count: 1
